architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,224 @@
+`Applications.Messaging` resources provider manages queues such as Rabbit MQ.
+
+Applications RP has a key sub component `Recipe Engine` to execute `recipes`. 
+`Recipes` are Bicep or Terraform code supplied by user that is used to deploy infrastructure components on Azure and AWS. The Bicep recipes are fetched from OCI compliant registries. Terraform recipes are public modules and fetched from internet too. 
+
+In order to execute Terraform recipes, Applications RP installs latest Terraform. It also mounts an empty directory `/terraform` into Applications RP pod. It uses this directory for executing terraform recipes using the installed executable. The output resources generated from terraform module are converted to Radius output resources and stored in our datastore. 
+
+In order to deploy bicep recipes, Applications RP sends a request to UCP, which in turn forwards it to Deployment Engine. 
+
+The RP uses a queue to process requests asyncronously. Information about resources that are deployed / being deployed is stored in a datastore. 
+
+Below is a high level overview of various key subcomponents in Applications RP
+![Applications RP](2024-10-applications-rp-threat-model/apprp.png) 
+
+
+### Implementation Details
+
+#### Use of Cryptography
+
+
+1. **Computing the Hash for Terraform Backend Configuration**: [Link to code](https://github.com/radius-project/radius/blob/main/pkg/recipes/terraform/config/backends/kubernetes.go#L110).
+
+   1. **Purpose**: The purpose of computing the hash is to compute a key for kubernetes secret which has the etrraform backend config
+   2. **Library**: The library used to calculate the hash of the deployment configuration is the crypto library, which is one of the standard libraries of Go: [Link to library](https://pkg.go.dev/crypto@go1.23.1).
+   3. **Type**: [SHA1](https://www.rfc-editor.org/rfc/rfc3174.html). Note: "SHA-1 is cryptographically broken and should not be used for secure applications." [Link to warning](https://pkg.go.dev/crypto/sha1@go1.23.1). This is used as an optimization for detecting changes, not as a security protection.
+
+2. Applications RP also uses UCP's queue and store libraries. These libraries use Cryptography and are detailed in UCP threat model. [link to be added]
+
+#### Storage of secrets
+
+Applications RP has access to sensitive information related to the application resource it manages as well as the cloud credentials it requires for managing cloud resources on Azure and AWS. Applications RP provides a Secret Store which can be used to store sensitive information such as TLS certificate and private keys. It uses kubernetes secrets to implement this secret store. 
```

Comment:
Feedback - split this into two big ideas

- Access to cloud credentials (do we store them?)
- Data that Applications RP needs to store.

BIG TOPIC - let's add AZWI/IRSA to this discussion.

Example:

Applications RP can access the cloud using either static credentials or OIDC Federation (AzWI/IRSA). We recommend that users configure federation where possible as no secrets are required.